File: packages/device_info/lib/src/flutter_config/src/models/src/optimization_result/optimization_result.dart
import 'package:device_info/device_info.dart';

class OptimizationResult {
  late FileInfo? originalPhotoWithNewPath;
  late FileInfo optimizedPhoto;
  late int savedSpaceInBytes;

  OptimizationResult({
    required this.originalPhotoWithNewPath,
    required this.optimizedPhoto,
    required this.savedSpaceInBytes,
  });
}

File: packages/device_info/android/src/main/kotlin/com/rocket/device/info/models/OptimizationResult.kt
package com.rocket.device.info.models

data class OptimizationResult(
    val originalPhotoWithNewPath: FileInfo?,
    val optimizedPhoto: FileInfo,
    val savedSpaceInBytes: Long,
){
    fun toMap(): Map<String, Any?>{
        return mapOf(
            "optimizedPhoto" to optimizedPhoto.toMap(),
            "originalPhotoWithNewPath" to originalPhotoWithNewPath?.toMap(),
            "savedSpaceInBytes" to savedSpaceInBytes
        )
    }
}
